src/ProjectNeon/Assets/VFX/RandomPlayAnimation.cs

```diff
@@ -0,0 +1,33 @@
+﻿using UnityEngine;
+using System.Collections.Generic;
+using System.Collections;
+
+public class RandomPlayAnimation : MonoBehaviour 
+{
+ 
+	public void Awake()
+    { 
+		this.gameObject.GetComponent<SpriteRenderer> ().enabled = false;
```

Comment:
There are still four calls to` GetComponent` in this class.

What you want to do is cache the component.

```

private SpriteRenderer _renderer;

public void Awake() 
{
    _renderer = GetComponent<SpriteRenderer>();
}
```

And then everytime you would get the component, you just use `_renderer.enabled`... etc.

Also, anytime you have a Script that requires another component in this way, it's best to put a RequiresComponent attribute on the Script like this: https://docs.unity3d.com/ScriptReference/RequireComponent.html